### ISSUE TITLE
isolated-declarations: TS9025 on an optional parameter property typed `readonly T[]`, which tsc 7 accepts

### ISSUE THREAD

**Issue by wagenet** [oxc-project/oxc#26944]
_Identified and written by Claude (Claude Code)._

oxc's isolated-declarations emitter rejects code that TypeScript 7 accepts. An optional parameter property typed `readonly string[]` fails with TS9025, while tsc 7 emits it with `| undefined` added, the same as it does for `string[]`.

## Reproduction

```ts
export class A {
  constructor(readonly p?: readonly string[]) {}
}
```

```js
import { isolatedDeclarationSync } from 'oxc-transform'; // 0.146.0

const r = isolatedDeclarationSync('a.ts', code, {});
console.log(r.errors.map((e) => e.message));
console.log(r.code);
```

oxc reports:

```
TS9025: Declaration emit for this parameter requires implicitly adding undefined to it's type. This is not supported with --isolatedDeclarations.
```

and emits the type without `| undefined`:

```ts
export declare class A {
	readonly p?: readonly string[];
	constructor(p?: readonly string[]);
}
```

`tsc` 7.0.2 (`--isolatedDeclarations --declaration --strict`) reports no errors and emits:

```ts
export declare class A {
    readonly p?: readonly string[] | undefined;
    constructor(p?: readonly string[] | undefined);
}
```

Change the type to `string[]` and oxc matches tsc: no error, `| undefined` added.

For reference, TypeScript 6.0.3 reports TS9025 here too, so oxc currently matches 6.x.

## Cause

In `crates/oxc_isolated_declarations/src/function.rs`, `undefined_presence` classifies `TSTypeOperatorType` as `Absent`, and `can_add_undefined` has no arm for it, so it falls to `_ => false`. `Absent` plus `false` hits the TS9025 branch.

`TSArrayType` is already in the `true` list of `can_add_undefined`, and adding `| undefined` after `readonly string[]` is just as syntactic. Adding `TSType::TSTypeOperatorType(_)` there looks like the fix. I tested this against `main` (591966d): it resolves the cases above and updates six existing snapshots in `default-parameter-type-kinds` and `function-parameters` that currently expect TS9025. tsc 7.0.2 reports no TS9025 on any of them, and the new output matches it, except for two `any`-typed cases where oxc adds a redundant `| undefined`.

## Related

With a type alias (`type P = 'x'; constructor(readonly p?: P)`), oxc emits `readonly p?: P` with no error, while tsc 7.0.2 emits `readonly p?: P | undefined`. That's the `Unresolved` path from #25909 and may be intentional, but the output differs.

## Impact

oxlint's `typescript/no-duplicate-type-constituents` (type-aware, via tsgolint) reports the explicit `| undefined` in `constructor(readonly p?: readonly string[] | undefined)` as unnecessary and autofixes it away. That's correct under TypeScript 7: tsc emits the same declaration with or without it, including with `exactOptionalPropertyTypes`. But the fixed code then fails oxc's isolated-declarations emit with TS9025, so following oxlint's autofix breaks declaration builds that use oxc (for example tsdown, which uses this emitter when `isolatedDeclarations` is on).

Reproduced with oxlint 1.85.0 + oxlint-tsgolint, oxc-transform 0.146.0, tsc 7.0.2.

## Versions

- oxc-transform 0.146.0 (same result via rolldown 1.2.5); the code path is unchanged on `main`
- oxlint 1.85.0 with oxlint-tsgolint
- TypeScript 7.0.2 for the comparison
- macOS arm64
